server/src/test/java/keywhiz/service/resources/BatchSecretDeliveryResourceTest.java

```diff
@@ -0,0 +1,163 @@
+            matches = true;
+        } else if (response.equals(ImmutableList.of(SecretDeliveryResponse.fromSecret(secret2), SecretDeliveryResponse.fromSecret(secret)))) {
+            matches = true;
+        }
+
+        assertThat(matches).isTrue();
+    }
+
+    @Test(expected = NotFoundException.class)
+    public void returnsNotFoundWhenClientDoesNotExist() throws Exception {
+        when(aclDAO.getBatchSanitizedSecretsFor(client, ImmutableList.of(secret.getName()))).thenReturn(ImmutableList.<SanitizedSecret>of());
+        when(clientDAO.getClientByName(client.getName())).thenReturn(Optional.empty());
+        when(secretController.getSecretsByName(ImmutableList.of(secret.getName())))
+                .thenReturn(ImmutableList.<Secret>of());
+
+        batchSecretDeliveryResource.getBatchSecret(client, BatchSecretRequest.create(ImmutableList.of(secret.getName())));
+    }
+
+
+    @Test(expected = NotFoundException.class)
+    public void returnsNotFoundWhenSecretDoesNotExist() throws Exception {
+        when(aclDAO.getBatchSanitizedSecretsFor(client, ImmutableList.of("secret_name"))).thenReturn(ImmutableList.<SanitizedSecret>of());
+        when(clientDAO.getClientByName(client.getName())).thenReturn(Optional.of(client));
+        when(secretController.getSecretsByName(ImmutableList.of("secret_name")))
+                .thenReturn(ImmutableList.<Secret>of());
+
+        batchSecretDeliveryResource.getBatchSecret(client, BatchSecretRequest.create(ImmutableList.of("secret_name")));
+    }
+
+
+    @Test(expected = ForbiddenException.class)
+    public void returnsUnauthorizedWhenDenied() throws Exception {
+        when(aclDAO.getBatchSanitizedSecretsFor(client, ImmutableList.of(secret.getName()))).thenReturn(ImmutableList.<SanitizedSecret>of());
+        when(clientDAO.getClientByName(client.getName())).thenReturn(Optional.of(client));
+        when(secretController.getSecretsByName(ImmutableList.of(secret.getName())))
+                .thenReturn(ImmutableList.<Secret>of(secret));
+
+        batchSecretDeliveryResource.getBatchSecret(client, BatchSecretRequest.create(ImmutableList.of(secret.getName())));
+    }
+
+    @Test
+    public void doesNotEscapeBase64() throws Exception {
+        String name = secretBase64.getName();
+
+        when(aclDAO.getBatchSanitizedSecretsFor(client, ImmutableList.of(name)))
+                .thenReturn(ImmutableList.of(SanitizedSecret.fromSecret(secretBase64)));
+        when(secretController.getSecretsByName(ImmutableList.of(name)))
+                .thenReturn(ImmutableList.of(secretBase64));
+
+        List<SecretDeliveryResponse> response = batchSecretDeliveryResource.getBatchSecret(client, BatchSecretRequest.create(ImmutableList.of(secretBase64.getName())));
+        assertThat(response.get(0).getSecret()).isEqualTo(secretBase64.getSecret());
+        assertThat(response.size()).isEqualTo(1);
```

Comment:
nit: I like to assert the list size first and retrieve items from it second, so that if it's empty the assertion fails rather than the test throwing an `IndexOutOfBoundsException`.